centreon/src/Core/Dashboard/Domain/Model/DashboardPanel.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Domain\Model;
+
+use Assert\AssertionFailedException;
+
+class DashboardPanel
+{
+    use DashboardPanelValidationTrait;
+    public const MAX_NAME_LENGTH = 200;
+    public const MAX_WIDGET_TYPE_LENGTH = 200;
+    public const MAX_WIDGET_SETTINGS_LENGTH = 65535;
+    public const MIN_SMALL_INTEGER = -32768;
+    public const MAX_SMALL_INTEGER = 32767;
+
+    private readonly string $name;
+
+    private readonly string $widgetType;
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param string $widgetType
+     * @param array<mixed> $widgetSettings
+     * @param int $layoutX
+     * @param int $layoutY
+     * @param int $layoutWidth
+     * @param int $layoutHeight
+     * @param int $layoutMinWidth
+     * @param int $layoutMinHeight
+     *
+     * @throws AssertionFailedException
+     */
+    public function __construct(
+        private readonly int $id,
+        string $name,
+        string $widgetType,
```

Comment:
There is no convention for properties not declared in constructor ? (e.g at the end or at the beginning)